ivy/functional/frontends/numpy/random/functions.py

```diff
@@ -233,3 +236,27 @@ def triangular(left, mode, right, size=None):
         right - (right - mode) * ((1 - u) * (right - mode) / (right - left)) ** 0.5
     )
     return ivy.where(condition, values1, values2)
+
+
+def hyper_helper(ngood, nbad, nsample):
+    u = []
+    for i in range(nsample):
+        choice = random.choices(
+            [1, 0], weights=(ngood / (ngood + nbad), nbad / (ngood + nbad)), k=1
+        )
+        u += choice
+        if choice == [1] and ngood > 0:
+            ngood -= 1
+        elif nbad > 0:
+            nbad -= 1
+    return sum(u)
+
+
+@to_ivy_arrays_and_back
+@from_zero_dim_arrays_to_scalar
+def hypergeometric(ngood, nbad, nsample, size=None):
+    u = ivy.empty(size, dtype=int)
```

Comment:
`size` argument is default None which is a required argument for `empty` which will always throw an error. As seen in the test logs [here](https://github.com/unifyai/ivy/actions/runs/5868789216/job/15912380408?pr=21922#step:3:247).

Reply:
Whenever I reproduce any of the failures I get an error that states that the test passed

Reply:
I tested the test cases locally and they don't fail like they do in the test suite. I even compared it to the numpy.random.hypergeometric module and they produce the same results (having in mind that the numbers are different due to the randomness)

Reply:
yea makes sense. The issue is a flaw in the logic. If you test call this function and don't specify a value for `size` then this would break because `None` is not a valid argument for `size` in `ivy.empty`.

Reply:
I dont get that issue in the new commits anymore

Reply:
Are all tests passing for you locally?

Reply:
locally I dont get any errors, yes

Reply:
I just want to mention, I have this many commits because my local tests only run for numpy and don't check the other backends.